examples_tests/44.LevelCurveExtraction/main.cpp

```diff
@@ -0,0 +1,235 @@
+            {
+                switch (shdrIn.type)
+                {
+                case asset::IPipelineMetadata::ECSI_WORLD_VIEW_PROJ:
+                {
+                    core::matrix4SIMD mvp = camera->getConcatenatedMatrix();
+                    memcpy(uboData.data()+shdrIn.descriptorSection.uniformBufferObject.relByteoffset, mvp.pointer(), shdrIn.descriptorSection.uniformBufferObject.bytesize);
+                }
+                break;
+                case asset::IPipelineMetadata::ECSI_WORLD_VIEW:
+                {
+                    core::matrix3x4SIMD MV = camera->getViewMatrix();
+                    memcpy(uboData.data() + shdrIn.descriptorSection.uniformBufferObject.relByteoffset, MV.pointer(), shdrIn.descriptorSection.uniformBufferObject.bytesize);
+                }
+                break;
+                case asset::IPipelineMetadata::ECSI_WORLD_VIEW_INVERSE_TRANSPOSE:
+                {
+                    core::matrix3x4SIMD MV = camera->getViewMatrix();
+                    memcpy(uboData.data()+shdrIn.descriptorSection.uniformBufferObject.relByteoffset, MV.pointer(), shdrIn.descriptorSection.uniformBufferObject.bytesize);
+                }
+                break;
+                }
+            }
+        }       
+        driver->updateBufferRangeViaStagingBuffer(gpuubo.get(), 0ull, gpuubo->getSize(), uboData.data());
+
+        for (uint32_t i = 0u; i < gpumesh->getMeshBufferCount(); ++i)
+        {
+            video::IGPUMeshBuffer* gpumb = gpumesh->getMeshBuffer(i);
+            const video::IGPURenderpassIndependentPipeline* pipeline = gpumb->getPipeline();  
+            const video::IGPUDescriptorSet* ds3 = gpumb->getAttachedDescriptorSet();
+
+            driver->bindGraphicsPipeline(pipeline);
+            const video::IGPUDescriptorSet* gpuds1_ptr = gpuds1.get();
+            driver->bindDescriptorSets(video::EPBP_GRAPHICS, pipeline->getLayout(), 1u, 1u, &gpuds1_ptr, nullptr);
+            const video::IGPUDescriptorSet* gpuds3_ptr = gpumb->getAttachedDescriptorSet();
+            if (gpuds3_ptr)
+                driver->bindDescriptorSets(video::EPBP_GRAPHICS, pipeline->getLayout(), 3u, 1u, &gpuds3_ptr, nullptr);
+            driver->pushConstants(pipeline->getLayout(), video::IGPUSpecializedShader::ESS_FRAGMENT, 0u, gpumb->MAX_PUSH_CONSTANT_BYTESIZE, gpumb->getPushConstantsDataPtr());
+
+            driver->drawMeshBuffer(gpumb);
+        }
+        //emit "memory barrier" of type GL_ALL_BARRIER_BITS after the entire scene finishes drawing
+        video::COpenGLExtensionHandler::extGlMemoryBarrier(GL_ALL_BARRIER_BITS);
+
+        
+        //invoke driver->drawIndirect() and use linesBuffer
+        driver->drawArraysIndirect(gpumesh->getMeshBuffer(0)->getVertexBufferBindings(), asset::EPT_LINE_LIST, linesBuffer.get(), 0u, triangleCount * 6 * sizeof(uint32_t), 6 * sizeof(uint32_t));
```

Comment:
wrong, you do not use the `gpumesh` for drawing lines, you need to provide your own array of 16 bindings

Reply:
`linesBuffer.get(), 0u, triangleCount * 6 * sizeof(uint32_t), 6 * sizeof(uint32_t)`
all of this is wrong, those parameters are for the buffer that holds `DrawIndirectParameters_t`